File: app/src/main/java/com/baihe/lihepro/activity/LoginActivity.java
package com.baihe.lihepro.activity;

import android.app.Application;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.baihe.common.base.BaseActivity;
import com.baihe.lihepro.R;

public class LoginActivity extends BaseActivity {
    public static final String IS_TO_HOME = "IS_TO_HOME";
    private boolean isToHome = true;
    public static LoginActivity instance;

    public static void start(Context context) {
        Intent starter = new Intent(context, LoginActivity.class);
        context.startActivity(starter);
    }

    public static void start(Context context, boolean isToHome) {
        Intent starter = new Intent(context, LoginActivity.class);
        starter.putExtra(IS_TO_HOME, isToHome);
        if(context instanceof Application){
            starter.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            starter.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        }
        context.startActivity(starter);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        instance = this;
        setContentView(R.layout.activity_login);
        if (getIntent() != null) {
            isToHome = getIntent().getBooleanExtra(IS_TO_HOME, true);
        }
    }

}

File: Lib_Login/src/main/java/com/baihe/lib_login/login/activity/LoginActivity.kt
package com.baihe.lib_login.login.activity

import android.app.Activity
import android.app.Application
import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.baihe.lib_framework.base.BaseActivity
import com.baihe.lib_login.R
import com.baihe.lib_login.constant.RequestCodeConstant

class LoginActivity: BaseActivity() {

    companion object{
        fun start(context: Context){
            val starter = Intent(context,LoginActivity::class.java)
            if (context is Application) {
                starter.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                starter.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                context.startActivity(starter)
                return
            }
            if (context is Activity){
                context.startActivityForResult(starter,RequestCodeConstant.REQUEST_LOGIN)
            }
        }
    }

    override fun getLayoutResId(): Int {
        return R.layout.login_activity_login
    }

    override fun initView(savedInstanceState: Bundle?) {

    }
}